File: pinyougou-portal-web/src/main/kotlin/com/pinyougou/portal/controller/ContentController.kt
package com.pinyougou.portal.controller

import com.alibaba.dubbo.config.annotation.Reference
import com.alibaba.dubbo.config.annotation.Service
import com.pinyougou.content.service.ContentService
import com.pinyougou.pojo.TbContent
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author 钟未鸣
 * @date 2018/4/15
 */
@RestController
@RequestMapping("/content")
class ContentController {
    @Reference
    lateinit var contentService: ContentService

    @RequestMapping("/findByCategoryId")
    fun findByCategoryId(categoryId: Long): List<TbContent> {
        return contentService.findByCategoryId(categoryId)
    }

}

File: pinyougou-portal-web/src/main/webapp/js/controller/contentController.js
app.controller("contentController",function($scope,contentService){
   $scope.findByCategoryId= function(categoryId){
       //存放不同目录的广告图片
        $scope.contentList=[];

       contentService.findByCategoryId(categoryId).success(
           function(response){
               //用id作为角标,把对应的数据存到对应角标
                $scope.contentList[categoryId] = response;
           }
       )
   }
});